src/net/connect.cpp

```diff
@@ -18,12 +19,9 @@ void mk_bufferevent_on_event(bufferevent *bev, short what, void *ptr) {
     if ((what & BEV_EVENT_CONNECTED) != 0) {
         (*cb)(mk::NoError(), bev);
     } else if ((what & BEV_EVENT_TIMEOUT) != 0) {
-        bufferevent_free(bev);
-        (*cb)(mk::net::TimeoutError(), nullptr);
+        (*cb)(mk::net::TimeoutError(), bev);
     } else {
-        bufferevent_free(bev);
-        // TODO: here we should map to the actual error that occurred
```

Comment:
I don't think this comment should be removed. It's here to remind us that we should portably access the error that occurred and accordingly map it onto our corresponding error.

Or, if you prefer the approach, we could modify `NetworkError` such that, when instantiated, it performs this operation. In that case the comment should probably be moved near `NetworkError`.